### PR TITLE
Added login system, class-creation, database,etc

### PR DESCRIPTION
**Pull request by rahuldugarrd**

### CODE REVIEW

File: login-system/classcreater.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+$_SESSION['classcode'] = $_POST['classcode'];
+$_SESSION['passcode'] = $_POST['passcode'];
```
Comment (pandeykartikey) [inline login-system/classcreater.php]: What is passcode?

File: login-system/createclass.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?php
+require 'db.php';
+session_start();
+?>
+<!DOCTYPE html>
+<html >
+<head>
+  <meta charset="UTF-8">
+  <title>Create new class</title>
+  <?php include 'css/css.html'; ?>
```
Comment (pandeykartikey) [inline login-system/createclass.php]: We don't include these files like this.
include as `something.css`.

File: login-system/createclass.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<body>
+    <div class="form">
+
+          <h1>Create new class</h1>
+          
+          <form action="createclass.php" method="post">
+              
+          <div class="field-wrap">
+            <label>
+              Class code(Preferably enter like "CSE2021",etc.<span class="req">*</span>
```
Comment (pandeykartikey) [inline login-system/createclass.php]: Use placeholders

File: login-system/db.php [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?php
+/* Database connection settings */
+$host = 'localhost';
+$user = 'root';
+$pass = 'passwd';
```
Comment (pandeykartikey) [inline login-system/db.php]: **You are exposing your password**
Use config.php as told on gitter.

File: login-system/error.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
+/* Displays all error messages */
+session_start();
+?>
+<!DOCTYPE html>
+<html>
+<head>
+  <title>Error</title>
+  <?php include 'css/css.html'; ?>
```
Comment (pandeykartikey) [inline login-system/error.php]: Here too.

File: login-system/index.php [hunk 6]
```diff
@@ -0,0 +1,132 @@
+<?php 
+/* Main page with two forms: sign up and log in */
+require 'db.php';
+session_start();
+?>
+<!DOCTYPE html>
+<html>
+<head>
+  <title>Sign-Up/Login Form</title>
+  <?php include 'css/css.html'; ?>
```
Comment (pandeykartikey) [inline login-system/index.php]: Here and everywhere below.

File: login-system/sql/sql_import.php [hunk 7]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+//connection variables
+$host = 'localhost';
+$user = 'root';
+$password = 'passwd';
```
Comment (pandeykartikey) [inline login-system/sql/sql_import.php]: you have defined db.php so why not include it here.

**Review by pandeykartikey — CHANGES_REQUESTED** [review]

**Comment by pandeykartikey** [comment]
@rahuldugarrd I have given it a general overview try to fix those things and try to break your code in smaller PR's because it is quite difficult to review.

File: login-system/classcreater.php (hunk 1 above)
Comment (rahuldugarrd) [reply login-system/classcreater.php]: passcode is to allow only students of that class to join.

File: login-system/createclass.php (hunk 3 above)
Comment (rahuldugarrd) [reply login-system/createclass.php]: These are like adaptive placeholders

File: login-system/sql/sql_import.php (hunk 7 above)
Comment (rahuldugarrd) [reply login-system/sql/sql_import.php]: these are the sql commands for creating database, tables, etc. and db.php is for connecting to database

File: login-system/index.php (hunk 6 above)
Comment (rahuldugarrd) [reply login-system/index.php]: Done...!

File: createclass.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+<html>
+<head>
+  <title>Course and Event Notification App</title>
+  <link rel="stylesheet" type="text/css" href="css/styles.css">
+</head>
+<body>
+  <div id="form">
+    <form action="classcreater.php" method="post">
+      <div id="input">
+        Title: <input type="text" name="title" placeholder="Enter title here..." id="inputs">
```
Comment (pandeykartikey) [inline createclass.php]: Use Label to associate an input with its heading

File: classcreater.php [hunk 9]
```diff
@@ -0,0 +1,23 @@
+<?php
+require 'config.php';
+$title = $_POST['title'];
+$classcode = $_POST['classcode'];
+$passcode = $_POST['passcode'];
+$q = $mysqli->query("SELECT * FROM class WHERE code = '$classcode'");
+if($q->num_rows>0)
+{
+  $_SESSION['message'] = 'class with this classcode already registered!';
+  header("location: displaymessage.php");
```
Comment (pandeykartikey) [inline classcreater.php:11]: Where is displaymessage.php?

File: signuppage.php [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+			<div id="input">
+				Password: <input type="password" name="password" placeholder="Enter password here..." id="inputs">
+			</div>
+			<div id="input">
+				Confirm password: <input type="password" name="confirmpassword" placeholder="Enter password again here..." id="inputs">
+			</div>
+			<div id="inputs">
+				<input type="submit" value="SignUp" id="submit">
+			</div>
+	</div>
```
Comment (pandeykartikey) [inline signuppage.php]: Add a link to login page.

File: css/styles.css [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+{
+	width: 100%;
+    padding: 5px 10px;
+    margin: 1px 0;
+    display: inline-block;
+    border: 1px solid #ccc;
+    border-radius: 4px;
+    box-sizing: border-box;
+}
+#submit
```
Comment (pandeykartikey) [inline css/styles.css]: Use classes for styling

File: css/styles.css [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    padding: 2px 20px;
+    margin: 5px 0px;
+    display: inline-block;
+    border: 1px solid #ccc;
+    border-radius: 4px;
+    box-sizing: border-box;
+}
+#inputs
+{
+	width: 100%;
```
Comment (pandeykartikey) [inline css/styles.css]: Fix indentation

File: sql/sql_import.php [hunk 13]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+$host = 'localhost';
+$user = 'root';
+$password = 'passwd';
```
Comment (pandeykartikey) [inline sql/sql_import.php]: Use config.php here also.
And add a sample config too :smile:

**Review by pandeykartikey — CHANGES_REQUESTED** [review]

**Comment by pandeykartikey** [comment]
Great Job :+1:

File: classcreater.php (hunk 9 above)
Comment (rahuldugarrd) [reply classcreater.php:11]: to be implemented for showing these messages

File: login.php [hunk 14]
```diff
@@ -1,12 +1,25 @@
 <?php
+session_start();
+require 'config.php';
 $email = $_POST['email'];
-$sql=$mysqli->query("SELECT * FROM users WHERE email = '$email'");
-if($sql->num_rows()>0)
+$q=$mysqli->query("SELECT * FROM users WHERE email = '$email'");
```
Comment (pandeykartikey) [inline login.php]: Keep more descriptive names

File: index.php [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+?>
+<!DOCTYPE html>
+<html>
+<head>
+	<title>Companion</title>
+	<link rel="stylesheet" type="text/css" href="css/styles.css">
+</head>
+<body>
+	<div id="form">
+      <form action="login.php" method="post">
```
Comment (pandeykartikey) [inline index.php]: Uneven Indentation

**Review by pandeykartikey — CHANGES_REQUESTED** [review]

File: index.php [hunk 16]
```diff
@@ -9,23 +9,26 @@
 	<link rel="stylesheet" type="text/css" href="css/styles.css">
 </head>
 <body>
-	<div id="form">
-      <form action="login.php" method="post">
-      <div id="input"> 
-        Username: <input type="email" name="email" placeholder="Username" id="inputs">
+	<div class="form">
```
Comment (pandeykartikey) [inline index.php]: still some indenting issues

File: index.php [hunk 17]
```diff
@@ -9,23 +9,26 @@
...
+    <form action="login.php" method="post">
+      <div class="input"> 
+        <label>Username: <input type="email" name="email" placeholder="Username" class="inputs">
```
Comment (pandeykartikey) [inline index.php]: Why email type input for username?

File: index.php [hunk 18]
```diff
@@ -9,23 +9,26 @@
...
+        </label>
       </div>
-      <div id="input">
-        Password: <input type="password" name="password" placeholder="password" id="inputs">
+      <div class="input">
+        <label>Password: <input type="password" name="password" placeholder="password" class="inputs">
```
Comment (pandeykartikey) [inline index.php]: Don't use labels like these. Look how they have been used here https://www.w3schools.com/tags/tag_label.asp

**Review by pandeykartikey — CHANGES_REQUESTED** [review]

**Comment by pandeykartikey** [comment]
You can also add a  `.gitignore` for `config.php` and add a `schema.sql` for the database structure (You can export this from PHPMyAdmin directly). 
You can add links to various pages of your website in `displaymessage.php` .

**Comment by pandeykartikey** [comment]
Do the aforesaid changes then we can merge this PR @rahuldugarrd

**Review by pandeykartikey — APPROVED** [review]

**Comment by pandeykartikey** [comment]
@rahuldugarrd now rebase on master and start work on a new branch. 
Make branch name more descriptive not like branch1 or mybranch.

**Pull request merged**